src/pybind/mgr/dashboard/controllers/rbd.py

```diff
@@ -1,136 +1,422 @@
-            return {'success': False, 'detail': str(e), 'errno': e.errno}
-        return {'success': True}
+
+        return _rbd_call(pool_name, _create)
+
+    @RbdTask('delete', ['{pool_name}', '{image_name}'], 2.0)
+    def delete(self, pool_name, image_name):
+        rbd_inst = rbd.RBD()
+        return _rbd_call(pool_name, rbd_inst.remove, image_name)
+
+    @RbdTask('edit', ['{pool_name}', '{image_name}'], 4.0)
+    @RESTController.args_from_json
+    def set(self, pool_name, image_name, name=None, size=None, features=None):
+        def _edit(ioctx, image):
+            rbd_inst = rbd.RBD()
+            # check rename image
+            if name and name != image_name:
+                rbd_inst.rename(ioctx, image_name, name)
+
+            # check resize
+            if size and size != image.size():
+                image.resize(size)
+
+            # check enable/disable features
+            if features is not None:
+                curr_features = _format_bitmask(image.features())
+                # check disabled features
+                _sort_features(curr_features, enable=False)
+                for feature in curr_features:
+                    if feature not in features and feature in self.ALLOW_DISABLE_FEATURES:
+                        f_bitmask = _format_features([feature])
+                        image.update_features(f_bitmask, False)
+                # check enabled features
+                _sort_features(features)
+                for feature in features:
+                    if feature not in curr_features and feature in self.ALLOW_ENABLE_FEATURES:
+                        f_bitmask = _format_features([feature])
+                        image.update_features(f_bitmask, True)
+
+        return _rbd_image_call(pool_name, image_name, _edit)
+
+    @RbdTask('copy',
+             {'src_pool_name': '{pool_name}',
+              'src_image_name': '{image_name}',
+              'dest_pool_name': '{dest_pool_name}',
+              'dest_image_name': '{dest_image_name}'}, 2.0)
+    @RESTController.resource(['POST'])
+    @RESTController.args_from_json
+    def copy(self, pool_name, image_name, dest_pool_name, dest_image_name,
+             obj_size=None, features=None, stripe_unit=None,
+             stripe_count=None, data_pool=None):
+
+        def _src_copy(s_ioctx, s_img):
+            def _copy(d_ioctx):
+                # Set order
+                l_order = None
+                if obj_size and obj_size > 0:
+                    l_order = int(round(math.log(float(obj_size), 2)))
+
+                # Set features
+                feature_bitmask = _format_features(features)
+
+                s_img.copy(d_ioctx, dest_image_name, feature_bitmask, l_order,
+                           stripe_unit, stripe_count, data_pool)
+
+            return _rbd_call(dest_pool_name, _copy)
+
+        return _rbd_image_call(pool_name, image_name, _src_copy)
+
+
+@ApiController('rbd/:pool_name/:image_name/snap')
```

Comment:
Nit: why not have all block-related things under a ```block``` sub-directory? iSCSI is ```block/iscsi```, this could be ```block/images/...```, and mirroring could be ```block/mirroring```.